src/fuse_concat.cpp

```diff
@@ -104,11 +98,37 @@ struct fused_concat
 MIGRAPHX_REGISTER_OP(fused_concat);
 
 namespace {
-struct find_concat_pointwise
+
+bool is_fusable_pointwise(instruction_ref ins)
+{
+    return ins->name() == "pointwise" and ins->outputs().size() == 1 and
+           ins->get_shape().type() != shape::tuple_type;
+}
+
+template <class... Ts>
+auto fusable_pointwise(Ts... xs)
+{
+    return match::name("pointwise")(match::not_tuple(), xs...);
+}
+
+template <std::size_t N, std::size_t Max = 2>
+struct concat_counter
+{
+    static_assert(N < Max, "Factor N must be less than Max");
+    std::shared_ptr<unsigned int> counter = std::make_shared<unsigned int>(0);
+    unsigned int get_noop_counter() const
+    {
+        if(counter == nullptr)
+            MIGRAPHX_THROW("Invalid counter");
```

Comment:
What could lead to this scenario?

Reply:
Currently there isnt a case but just checking in case there is a future case where this would fail.